src/creator.rs

```diff
@@ -2,16 +2,33 @@
 
 extern crate pwasm_std;
 extern crate pwasm_ethereum;
+extern crate bigint;
+extern crate parity_hash as hash;
 
 use pwasm_std::logger;
-use pwasm_ethereum::{input, ret, create, value};
+use pwasm_ethereum::{input, ret, create, create2, value};
+use bigint::U256;
+use hash::H256;
 
 #[no_mangle]
 pub fn call() {
-	if let Ok(addr) = create(value(), &input()) {
-		logger::debug("Created contractwith code");
-		ret(&addr[..]);
+	let mut r = [0u8; 40];
+	if let Ok(addr) = create(value() / U256::from(2), &input()) {
+		logger::debug("Created contract with code");
+		(&mut r[0..20]).copy_from_slice(&addr[..]);
 	} else {
 		logger::debug("Error creating contract");
 	}
+
+	if let Ok(addr) = create2(value() / U256::from(2), H256::default(), &input()) {
```

Comment:
maybe let's use some non-zero salt to actually test the argument reaches the runtime after?